modules/eloqua_api_auth_fallback/src/Form/Settings.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Drupal\eloqua_api_auth_fallback\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Eloqua API Fallback Authentication Settings.
+ */
+class Settings extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'eloqua_api_auth_fallback_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['eloqua_api_auth_fallback.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('eloqua_api_auth_fallback.settings');
+
+    $form['credentials'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Eloqua Instance Authentication Credentials'),
+    ];
+
+    $form['credentials']['sitename'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Instance/Site Name'),
+      '#default_value' => $config->get('sitename'),
+    ];
+
+    $form['credentials']['username'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Username'),
+      '#default_value' => $config->get('username'),
+    ];
+
+    $form['credentials']['password'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Password'),
+      '#default_value' => $config->get('password'),
```

Comment:
Might be a good idea to add a `#description`

Reply:
Given all 3 values are required as per the [API docs](https://docs.oracle.com/cloud/latest/marketingcs_gs/OMCAB/Developers/GettingStarted/Authentication/authenticate-using-oauth.htm#resource-owner-password-credentials-grant)..
might be a good idea to set `'#required' => TRUE`